rust/pylance/src/scanner.rs

```diff
@@ -0,0 +1,93 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use pyo3::prelude::*;
+use pyo3::{pyclass, PyObject, PyResult};
+/// This a Scanner implementation which will being associated
+/// with a Pyarrow Scanner so duckdb can be tricked to call
+/// it's methods during scans
+use std::sync::Arc;
+
+use arrow::ffi_stream::*;
+use arrow::pyarrow::*;
+use arrow_schema::Schema as ArrowSchema;
+
+use crate::errors::ioerror;
+use crate::reader::LanceReader;
+use ::lance::dataset::scanner::Scanner as LanceScanner;
+use ::lance::dataset::Dataset as LanceDataset;
+use pyo3::exceptions::PyValueError;
+use tokio::runtime::Runtime;
+
+#[pyclass(name = "_Scanner", module = "_lib")]
+pub struct Scanner {
+    dataset: Arc<LanceDataset>,
+    columns: Option<Vec<String>>,
+    offset: Option<i64>,
+    limit: i64,
+    rt: Arc<Runtime>,
+}
+
+impl Scanner {
+    pub fn new(
+        dataset: Arc<LanceDataset>,
+        columns: Option<Vec<String>>,
+        offset: Option<i64>,
+        limit: i64,
+        rt: Arc<Runtime>,
+    ) -> Self {
+        Self {
+            dataset,
+            columns,
+            offset,
+            limit,
+            rt,
+        }
+    }
+}
+
+#[pymethods]
+impl Scanner {
+    #[getter(dataset_schema)]
+    fn dataset_schema(self_: PyRef<'_, Self>) -> PyResult<PyObject> {
```

Comment:
why use `self_`  as name?

Reply:
Convention from examples - this has a reference to Python whereas &self: Self does not